sys/include/atomic_utils.h

```diff
@@ -712,14 +739,10 @@ static inline void semi_atomic_fetch_and_u64(uint64_t *dest, uint64_t val);
  * @param   type    Variable type, e.g. `uint8_t`
  */
 #define ATOMIC_LOAD_IMPL(name, type)                                           \
-    static inline type CONCAT(atomic_load_, name)(const type *var)             \
+    static inline type CONCAT(atomic_load_, name)(const volatile type *var)    \
     {                                                                          \
         unsigned state = irq_disable();                                        \
-        /* var can be register allocated, hence the memory barrier of          \
-         * irq_disable() and irq_restore() may not apply here. Using volatile  \
-         * ensures that the compiler allocates it in memory and that the       \
-         * memory access is not optimized out. */                              \
-        type result = *((const volatile type *)var);                           \
```

Comment:
Maybe I should have done this right from the beginning just for this simplification.